tests/lib/cli-engine/cli-engine.js

```diff
@@ -3454,6 +3454,46 @@ describe("CLIEngine", () => {
                 assert.strictEqual(report.results[0].messages[0].message, "ok");
             });
         });
+
+        describe("glob pattern '[ab].js'", () => {
+            const root = getFixturePath("cli-engine/unmatched-glob");
+
+            it("should match 'a.js' and 'b.js', but not '[ab].js'.", () => {
+                CLIEngine = defineCLIEngineWithInMemoryFileSystem({
+                    cwd: () => root,
+                    files: {
+                        "a.js": "",
+                        "b.js": "",
+                        "ab.js": "",
+                        "[ab].js": "",
+                        ".eslintrc.yml": "root: true"
+                    }
+                }).CLIEngine;
+                engine = new CLIEngine();
+
+                const { results } = engine.executeOnFiles(["[ab].js"]);
+                const filenames = results.map(r => path.basename(r.filePath));
+
+                assert.deepStrictEqual(filenames, ["a.js", "b.js"]);
```

Comment:
what if the user want to match "[ab].js" in this case?

just my 2c. :-) it seems more straightforward like this:
1. '[ab].js' as a filename
2. '[ab].js' as a glob pattern (if the file not exist).

Reply:
At first, I thought so, but it's different from what shell does.

```
root@387125dc75aa:~# ls
[ab].js  a.js  b.js
root@387125dc75aa:~# echo [ab].js
a.js b.js
```

I think that it's better if the behavior is similar to popular tools.

Reply:
the question is the user cannot lint "[ab].js", even quoting `eslint "[ab].js"`.

I agreed we should be as close to popular tools as possible, but I'm not sure if we can make an exception here.

Reply:
Yes. It's the problem that ESLint cannot distinguish `eslint "foo"` and `eslint foo`.

I'm considering that Windows has the same behavior with Linux with `eslint [ab].js`. Hmm.